python_modules/dagster-test/dagster_test/graph_job_op_toys/input_managers.py

```diff
@@ -0,0 +1,118 @@
+import os
+
+import numpy as np
+import pandas as pd
+import pyspark.sql as pyspark
+
+from dagster import (
+    Field,
+    IOManager,
+    In,
+    InputContext,
+    Noneable,
+    OutputContext,
+    graph,
+    io_manager,
+    op,
+)
+
+
+class PandasCsvIOManager(IOManager):
+    def __init__(self, base_dir=None):
+        self.base_dir = os.getenv("DAGSTER_HOME") if base_dir is None else base_dir
+
+    def _get_path(self, output_context: OutputContext):
+        return os.path.join(
+            self.base_dir, "storage", f"{output_context.step_key}_{output_context.name}.csv"
+        )
+
+    def handle_output(self, context: OutputContext, obj: pd.DataFrame):
+        file_path = self._get_path(context)
+        os.makedirs(os.path.dirname(file_path), exist_ok=True)
+        if obj is not None:
+            obj.to_csv(file_path, index=False)
+
+    def load_input(self, context: InputContext) -> pd.DataFrame:
+        return pd.read_csv(self._get_path(context.upstream_output))  # type: ignore
+
+
+@io_manager(config_schema={"base_dir": Field(Noneable(str), default_value=None, is_required=False)})
+def pandas_io_manager(init_context):
+    return PandasCsvIOManager(base_dir=init_context.resource_config["base_dir"])
+
+
+class PysparkCsvInputManager(PandasCsvIOManager):
+    def load_input(self, context: InputContext) -> pyspark.DataFrame:
+        file_path = self._get_path(context.upstream_output)
+        spark = pyspark.SparkSession.builder.getOrCreate()
+        df = spark.read.csv(file_path)
+        return df
+
+
+@io_manager(config_schema={"base_dir": Field(Noneable(str), default_value=None, is_required=False)})
+def pyspark_input_manager(init_context):
```

Comment:
had issues passing pyspark dataframe around (never used pyspark before so i don't really know what im doing with it) so switched to using numpy for the sake of getting the example working. keeping the pyspark io manager around just in case i want to use it again, but you can ignore it for now